enhancements/installer/vsphere-ipi-zonal.md

```diff
@@ -0,0 +1,1067 @@
+datacenters and clusters.
+
+### Workflow Description
+
+### api
+
+#### Infrastructure
+
+Using the upstream [vSphere cluster api](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/apis/v1beta1/vspherefailuredomain_types.go)
+and the upstream [vSphere cloud config manager](https://github.com/kubernetes/cloud-provider-vsphere/blob/master/pkg/common/config/types_yaml.go)
+as examples to implement parameters of `VSpherePlatformSpec`.
+These parameters include the optional and required information to manage a OpenShift cluster on vSphere.
+
+```golang
+// VSpherePlatformFailureDomainType is the name of the failure domain type.
+// There are two defined failure domains currently, Datacenter and ComputeCluster.
+// Each represents a vCenter object type within a vSphere environment.
+// +kubebuilder:validation:Enum=HostGroup;Datacenter;ComputeCluster
+// + ---
+// + HostGroup is included for future support, consumers should output an error
+// + if this value is set.
+type VSpherePlatformFailureDomainType string
+
+const (
+	// HostGroupFailureDomain as a type allows the use of a group of ESXi hosts
+	// to be represented as a failure domain zone. When using this
+	// case it is expected that region would be a cluster.
+	// HostGroups within vCenter must be preconfigured and
+	// assigned in the topology.
+	HostGroupFailureDomain VSpherePlatformFailureDomainType = "HostGroup"
+
+	// ComputeClusterFailureDomain failure domain can either be a zone or region.
+	// The vCenter cluster is required to preconfigured and
+	// assigned in the topology.
+	ComputeClusterFailureDomain VSpherePlatformFailureDomainType = "ComputeCluster"
+
+	// DatacenterFailureDomain failure domain can be only be a region. The vcenter
+	// datacenter is required to be preconfigred and assigned
+	// in the topology. If used the zone would be of type ComputeCluster.
+	DatacenterFailureDomain VSpherePlatformFailureDomainType = "Datacenter"
+)
+
+// VSpherePlatformFailureDomainSpec holds the region and zone failure domain and
+// the vCenter topology of that failure domain.
+type VSpherePlatformFailureDomainSpec struct {
+	// name defines the name of the VSpherePlatformFailureDomainSpec
+	// This name is arbitrary but will be used
+	// in VSpherePlatformDeploymentZone for association.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	Name string `json:"name"`
+
+	// region defines a VSpherePlatformFailureDomain which
+	// includes the name of the vCenter tag, the failure domain type
+	// and the name of the vCenter tag category.
+	// +kubebuilder:validation:Required
+	Region VSpherePlatformFailureDomain `json:"region"`
+
+	// zone defines a VSpherePlatformFailureDomain which
+	// includes the name of the vCenter tag, the failure domain type
+	// and the name of the vCenter tag category.
+	// +kubebuilder:validation:Required
+	Zone VSpherePlatformFailureDomain `json:"zone"`
```

Comment:
Will these always be identical constructs? As in, should they share the same underling type?

Reply:
Removed with api changes